opencti-platform/opencti-graphql/src/http/httpPlatform.js

```diff
@@ -51,6 +51,19 @@ const extractRefererPathFromReq = (req) => {
   return undefined;
 };
 
+const publishFileDownload = async (executeContext, auth, file) => {
+  const { filename, entity_id } = file.metaData;
+  const entity = entity_id ? await internalLoadById(executeContext, auth, entity_id) : undefined;
+  const data = buildContextDataForFile(entity, file.id, filename);
+  await publishUserAction({
+    user: auth,
+    event_type: 'file',
+    event_access: 'extended',
+    event_scope: 'download',
+    context_data: data
+  });
+};
+
```

Comment:
Can't we just have one function taking the scope `download` or `read` in parameter to avoid duplicates ? In a same way `checkFileAccess` has been made

```suggestion
const publishFile = async (executeContext, auth, file, event_scope) => {
  const { filename, entity_id } = file.metaData;
  const entity = entity_id ? await internalLoadById(executeContext, auth, entity_id) : undefined;
  const data = buildContextDataForFile(entity, file.id, filename);
  await publishUserAction({
    user: auth,
    event_type: 'file',
    event_access: 'extended',
    event_scope,
    context_data: data
  });
};

```

Reply:
Thanks for the comment. I will merge as it is and will be part of future refactor.